functions/webhooks/twitter/FlexChannelUpdate.protected.ts

```diff
@@ -0,0 +1,80 @@
+import '@twilio-labs/serverless-runtime-types';
+import { Context, ServerlessCallback } from '@twilio-labs/serverless-runtime-types/types';
+import {
+  responseWithCors,
+  bindResolve,
+  error400,
+  error500,
+  success,
+} from '@tech-matters/serverless-helpers';
+
+type EnvVars = {
+  CHAT_SERVICE_SID: string;
+};
+
+export type Body = {
+  Source?: string;
+  ChannelSid?: string;
+  Attributes?: string; // channel attributes (e.g. "{\"from\":\"pgian\",\"channel_type\":\"custom\",\"status\":\"INACTIVE\",\"long_lived\":false}")
+  UniqueName?: string;
+  FriendlyName?: string;
+  ClientIdentity?: string; // client firing the channel update
+  CreatedBy?: string;
+  EventType?: string;
+  InstanceSid?: string;
+  DateCreated?: string;
+  DateUpdated?: string;
+  AccountSid?: string;
+  RetryCount?: string;
+  WebhookType?: string;
+  ChannelType?: string;
+  WebhookSid?: string;
+};
+
+function timeout(ms: number) {
+  return new Promise(resolve => setTimeout(resolve, ms));
+}
+
+export const handler = async (
+  context: Context<EnvVars>,
+  event: Body,
+  callback: ServerlessCallback,
+) => {
+  const response = responseWithCors();
+  const resolve = bindResolve(callback)(response);
+
+  try {
+    console.log('------ FlexChannelUpdate excecution ------');
+    const client = context.getTwilioClient();
+
+    if (event.EventType === 'onChannelUpdated') {
+      const { ChannelSid } = event;
+
+      if (ChannelSid === undefined) {
+        resolve(error400('ChannelSid'));
+        return;
+      }
+
+      const channel = await client.chat
+        .services(context.CHAT_SERVICE_SID)
+        .channels(ChannelSid)
+        .fetch();
+
+      const { status } = JSON.parse(channel.attributes);
+
+      if (status === 'INACTIVE') {
+        await timeout(1000); // set small timeout just in case some cleanup is still going on
+        const removed = await channel.remove();
+        console.log(`INACTIVE channel with sid ${ChannelSid} removed: ${removed}`);
+        resolve(success(`INACTIVE channel removed: ${removed}`));
+        return;
```

Comment:
I think the name of the function could be more descriptive of what it actually does. Something like `RemoveInactiveFlexChannel.protected.ts`?

Reply:
I'm about to change the behavior of this webhook, cause when doing this we lose the chat history in Twilio. The idea is to preserve the channels and keep track a map of Twitter user - active channel.